### PR TITLE
Otabek

### PR DESCRIPTION
**Pull request by Otabekkadirov**
Here I confirm that I:

- [x] ran `yarn test` before pushing changes.
- [x] wrote myself and understood all the code.
- [x] checked cleanliness of my code after finishing the implementation.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 import menu from "./menu.js";
 
 // TODO
+menu.addDishToCourse('mains', 'Palov', 35);
```
Comment (bahriddin) [inline index.js]: It is not finished yet.

File: menu.js [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+      name: dishName,
+      price: dishPrice
+    };
+    if (courseName === 'appetizers') {
+      menu._courses.appetizers.push(dish);
+    } else if (courseName === 'mains') {
+      menu._courses.mains.push(dish);
+    } else if (courseName === 'desserts') {
+      menu._courses.desserts.push(dish);
+    }
```
Comment (bahriddin) [inline menu.js]:
```suggestion
    this._courses[courseName].push(dish);
```

File: menu.js [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+  },
+
+  getRandomDishFromCourse(courseName) {
+    let mealIndex = Math.floor(Math.random() * this._courses[courseName].length);
+    return this._courses[courseName][mealIndex];
```
Comment (bahriddin) [inline menu.js]: You are calling `this._courses[courseName]` 2 times. I think making extra variable for its value is a good idea. Something like `courseDishes` should be fine.

File: menu.js [hunk 4]
```diff
@@ -1,5 +1,48 @@
...
+  },
+
+  generateRandomMeal() {
+    const appetizer = this.getRandomDishFromCourse('appetizers');
+    const main = this.getRandomDishFromCourse('mains');
+    const dessert = this.getRandomDishFromCourse('desserts');
+
+    const dishes = [appetizer.name, main.name, dessert.name];
+    const total = appetizer.price + main.price + dessert.price;
```
Comment (bahriddin) [inline menu.js]: It seems we don't need these 2 variables as they are 
- used only once
- do not add great value to our code readability.

File: mystery.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (bahriddin) [inline mystery.js]: What is this? 🕵🏼‍♂️

**Review by bahriddin — CHANGES_REQUESTED** [review]

File: mystery.js (hunk 5 above)
Comment (Otabekkadirov) [reply mystery.js]: oh i forgot to delete this file🤦🏻‍♂️

File: menu.js (hunk 4 above)
Comment (Otabekkadirov) [reply menu.js]: I couldn't get the total price with a $ sign any other way, now I think I figured it out.

File: menu.js (hunk 3 above)
Comment (Otabekkadirov) [reply menu.js]: At first it was as you suggested...

File: index.js (hunk 1 above)
Comment (Otabekkadirov) [reply index.js]: So sorry, it is all because of leaderboard rush)))))

Comment (bahriddin) [reply index.js]: Quality should be no. 1 priority.

File: index.js [hunk 6]
```diff
@@ -1,3 +1,13 @@
...
+menu.addDishToCourse('mains', 'Beef T-Bone Steak', 15);
+menu.addDishToCourse('desserts', 'Chocolate chip cookies', 3);
+
+menu.addDishToCourse('appetizers', 'Caprese Garlic Bread', 8);
+menu.addDishToCourse('mains', 'Lasagna', 13);
+menu.addDishToCourse('desserts', 'Banana pudding', 5);
+
+menu.addDishToCourse('appetizers', 'Oven Baked Zucchini Chips ', 8);
+menu.addDishToCourse('mains', 'Creamy Tomato Pasta', 10);
+menu.addDishToCourse('desserts', 'Tiramisu', 4);
```
Comment (bahriddin) [inline index.js]: It is still not finished. **README** file is there for you to read.

**Review by bahriddin — CHANGES_REQUESTED** [review]

**Comment by Otabekkadirov** [comment]
Bahriddin aka, README faylda aytilganidek qo'shdim. Lekin kommentga olib qo'ygan kodim ham ishlaydimi shu joyda?

**Comment by bahriddin** [comment]
> Bahriddin aka, README faylda aytilganidek qo'shdim. Lekin kommentga olib qo'ygan kodim ham ishlaydimi shu joyda?

No. In general, this file is only for your own running `node index.js`.

File: menu.js [hunk 7]
```diff
@@ -1,5 +1,33 @@
...
+  },
+  generateRandomMeal() {
+    const appetizer = this.getRandomDishFromCourse('appetizers');
+    const main = this.getRandomDishFromCourse('mains');
+    const dessert = this.getRandomDishFromCourse('desserts');
+    const dishNameAndTotalPrice = {
+      dishes: [appetizer.name, main.name, dessert.name],
+      total: '$' + [appetizer.price + main.price + dessert.price]
+    }
+    return dishNameAndTotalPrice;
```
Comment (bahriddin) [inline menu.js]:
```suggestion
    return {
      dishes: [appetizer.name, main.name, dessert.name],
      total: '$' + (appetizer.price + main.price + dessert.price)
    };
```

**Review by bahriddin — CHANGES_REQUESTED** [review]

File: index.js [hunk 8]
```diff
@@ -1,3 +1,25 @@
...
+menu.addDishToCourse('mains', 'Ravioli with Coconut Milk & Lemongrass', 10);
+menu.addDishToCourse('desserts', 'Strawberry Ice-cream', 3);
+
+menu.addDishToCourse('appetizers', 'Fresh salad with tomato & cucumber', 5);
+menu.addDishToCourse('mains', 'Uzbek Homa-made Palov', 12);
+menu.addDishToCourse('desserts', 'Turkish Baklava', 4);
+
+const meal = menu.generateRandomMeal();
+console.log(meal);
+// console.log(menu.generateRandomMeal());
```
Comment (bahriddin) [inline index.js]: One last change: remove this commented code and you're done! 😄

**Review by bahriddin — CHANGES_REQUESTED** [review]

File: index.js (hunk 8 above)
Comment (bahriddin) [reply index.js]: **Rule of thumb** is when you are requesting code review, only working part of your code should be kept.

**Review by bahriddin — APPROVED** [review]
👍

**Pull request closed without merging**